### PR TITLE
Add select parameter to API endpoints

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/16716 |

## Description

Added the `select` parameter to several endpoints, updated their tests and specifications.

## Tests

```console
(venv) gasti@pop-os:~/work/wazuh$ python3 -m pytest api/api --disable-warnings
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.16, pytest-5.4.3, py-1.11.0, pluggy-0.13.1
rootdir: /home/gasti/work/wazuh/api
plugins: aiohttp-0.3.0, anyio-3.6.2, tavern-1.2.2, trio-0.7.0, html-2.1.1, asyncio-0.15.1, metadata-2.0.4
collected 544 items                                                                                                                                                                          

api/api/controllers/test/test_active_response_controller.py .                                                                                                                          [  0%]
api/api/controllers/test/test_agent_controller.py ...........................................                                                                                          [  8%]
api/api/controllers/test/test_cdb_list_controller.py ......                                                                                                                            [  9%]
api/api/controllers/test/test_ciscat_controller.py .                                                                                                                                   [  9%]
api/api/controllers/test/test_cluster_controller.py ........................                                                                                                           [ 13%]
api/api/controllers/test/test_decoder_controller.py .......                                                                                                                            [ 15%]
api/api/controllers/test/test_default_controller.py .                                                                                                                                  [ 15%]
api/api/controllers/test/test_experimental_controller.py ...............                                                                                                               [ 18%]
api/api/controllers/test/test_manager_controller.py ..................                                                                                                                 [ 21%]
api/api/controllers/test/test_mitre_controller.py .......                                                                                                                              [ 22%]
api/api/controllers/test/test_overview_controller.py .                                                                                                                                 [ 22%]
api/api/controllers/test/test_rootcheck_controller.py ....                                                                                                                             [ 23%]
api/api/controllers/test/test_rule_controller.py ........                                                                                                                              [ 25%]
api/api/controllers/test/test_sca_controller.py ..                                                                                                                                     [ 25%]
api/api/controllers/test/test_security_controller.py ...................................................                                                                               [ 34%]
api/api/controllers/test/test_syscheck_controller.py ....                                                                                                                              [ 35%]
api/api/controllers/test/test_syscollector_controller.py .........                                                                                                                     [ 37%]
api/api/controllers/test/test_task_controller.py .                                                                                                                                     [ 37%]
api/api/controllers/test/test_vulnerability_controller.py ....                                                                                                                         [ 38%]
api/api/models/test/test_model.py ...........................                                                                                                                          [ 43%]
api/api/test/test_alogging.py ..................                                                                                                                                       [ 46%]
api/api/test/test_authentication.py ...........                                                                                                                                        [ 48%]
api/api/test/test_configuration.py ............................................                                                                                                        [ 56%]
api/api/test/test_encoder.py ...                                                                                                                                                       [ 56%]
api/api/test/test_middlewares.py ............                                                                                                                                          [ 59%]
api/api/test/test_uri_parser.py ...                                                                                                                                                    [ 59%]
api/api/test/test_util.py ..............................................                                                                                                               [ 68%]
api/api/test/test_validator.py ....................................................................................................................................................... [ 95%]
......................                                                                                                                                                                 [100%]

============================================================================== 544 passed, 21 warnings in 2.21s ==============================================================================

```

### CODE REVIEW

**Review by nico-stefani — DISMISSED** [review]
LGTM!

**Review by Selutario — CHANGES_REQUESTED** [review]
Also, add new AITs to review the correct operation of the `select` field in the endpoints it was added.

File: api/test/integration/test_cluster_endpoints.tavern.yaml [hunk 1]
```diff
@@ -1778,6 +1778,37 @@ stages:
...
+          select_key: 'timestamp,tag'
+      status_code: 200
+      json:
+        error: 0
+        data:
+          total_affected_items: !anyint
+          failed_items: []
+          total_failed_items: 0
+
+  - name: Read logs using invalid select
```
Comment (Selutario) [inline api/test/integration/test_cluster_endpoints.tavern.yaml:1801]:
```suggestion
  - name: Try to read logs using invalid select
```

File: api/test/integration/test_decoder_endpoints.tavern.yaml [hunk 2]
```diff
@@ -731,6 +731,37 @@ stages:
...
+          select_key: 'filename,status' # required_fields={'filename'}
+      status_code: 200
+      json:
+        error: 0
+        data:
+          total_affected_items: !anyint
+          failed_items: []
+          total_failed_items: 0
+
+  - name: Show decoders using invalid select
```
Comment (Selutario) [inline api/test/integration/test_decoder_endpoints.tavern.yaml:754]: Same as above, you will need to review this in all the new tests.

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — APPROVED** [review]
LGTM, moving this to blocked until https://github.com/wazuh/wazuh/pull/17703 is merged and this branch rebased.

**Comment by Selutario** [comment]
Done in https://github.com/wazuh/wazuh/pull/17865

**Pull request closed without merging**